python/semantic_kernel/utils/model_diagnostics.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) Microsoft. All rights reserved.
+#
+# Model diagnostics to trace model activities with the OTel semantic conventions.
+# This code contains experimental features and may change in the future.
+# To enable these features, set one of the following senvironment variables to true:
+#    SEMANTICKERNEL_EXPERIMENTAL_GENAI_ENABLE_OTEL_DIAGNOSTICS
+#    SEMANTICKERNEL_EXPERIMENTAL_GENAI_ENABLE_OTEL_DIAGNOSTICS_SENSITIVE
+
+import json
+import os
+from typing import Optional
+
+from azure.monitor.opentelemetry import configure_azure_monitor
+from opentelemetry import trace
+from opentelemetry.trace import Span
+
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.contents import ChatHistory
+from semantic_kernel.contents.chat_message_content import ITEM_TYPES, ChatMessageContent
+from semantic_kernel.contents.function_call_content import FunctionCallContent
+
+# Activity tags
+_SYSTEM = "gen_ai.system"
+_OPERATION = "gen_ai.operation.name"
+_MODEL = "gen_ai.request.model"
+_MAX_TOKEN = "gen_ai.request.max_tokens"
+_TEMPERATURE = "gen_ai.request.temperature"
+_TOP_P = "gen_ai.request.top_p"
+_RESPONSE_ID = "gen_ai.response.id"
+_FINISH_REASON = "gen_ai.response.finish_reason"
+_PROMPT_TOKEN = "gen_ai.response.prompt_tokens"
+_COMPLETION_TOKEN = "gen_ai.response.completion_tokens"
+
+# Activity events
+PROMPT_EVENT_PROMPT = "gen_ai.prompt"
+COMPLETION_EVENT_COMPLETION = "gen_ai.completion"
+
+_enable_diagnostics = os.getenv("SEMANTICKERNEL_EXPERIMENTAL_GENAI_ENABLE_OTEL_DIAGNOSTICS",
```

Comment:
I think we can follow the pattern of telemetry here (telemetry.py in the connectors folder), where the relevant output (ENABLE_DIAGNOSTICS and ENABLE_SENSITIVE_EVENTS) are exposed as constants.

Reply:
Done